### PR TITLE
Error code

### PR DESCRIPTION
**Pull request by calvera**
Now it allows developer to return error codes from Exceptions

### CODE REVIEW

File: src/Error/Error.php [hunk 1]
```diff
@@ -124,9 +128,9 @@ public static function formatError(Error $error)
      * @param array|null $path
      * @param \Exception|\Error $previous
      */
-    public function __construct($message, $nodes = null, Source $source = null, $positions = null, $path = null, $previous = null)
+    public function __construct($message, $nodes = null, Source $source = null, $positions = null, $path = null, $previous = null, $code = null)
     {
-        parent::__construct($message, 0, $previous);
+        parent::__construct($message, $code ?: 0, $previous);
```
Comment (vladar) [inline src/Error/Error.php]: I don't think it's worth adding `$code` argument to `GraphQL\Error\Error`. This exception is for internal library usage and we don't use exception codes in the library.

Instead, I suggest to include the code in formatted error only if there is a previous exception and this exception has a code. 

So the only place which should be adjusted is `GraphQL\Error\FormattedError`

Comment (calvera) [reply src/Error/Error.php]: so it is misused in overblog/graphqlbundle? @overblog uses it in errorhandler

Comment (vladar) [reply src/Error/Error.php]: I am not familiar with overblog bundle as I don't use Symfony, but if you point me the line where it is misused, I'll take a look

Comment (calvera) [reply src/Error/Error.php]: as i understand your previous message, it is not supposed that anybody uses GraphQL\Error\Error. so https://github.com/overblog/GraphQLBundle/blob/0b3b90ec54f06c7cc686962b3528fc0aae0b2e5e/Error/ErrorHandler.php#L115 and https://github.com/overblog/GraphQLBundle/blob/0b3b90ec54f06c7cc686962b3528fc0aae0b2e5e/Error/ErrorHandler.php#L127 is therefore misuse...

those lines are the reason i propose my PR

thanks!

Comment (vladar) [reply src/Error/Error.php]: Well looks like they just replace a message of some exceptions to "Internal server error". They don't use exception code for `GraphQL\Error\Error` as well. The way they use this class is aligned with the way how we use it in this library.

I still don't quite understand why do you need code for base exception vs just using the code of the previous one?

Comment (calvera) [reply src/Error/Error.php]: yes, they do not use code, but i would like to throw exception with code in my resolver and have the code in the error
after the changes i propose, i get it like this:
```yml
{
    "data": {
        "checkOut": null
    },
    "errors": [
        {
            "message": "Cannot check out, not checked in",
            "code": 1052,
            "locations": [
                {
                    "line": 2,
                    "column": 3
                }
            ],
            "path": [
                "checkOut"
            ]
        }
    ]```

Comment (vladar) [reply src/Error/Error.php]: I do understand this. And you can access this code from `$error->getPrevious()->getCode();` right in error formatter. You don't need to add $code argument to constructor of `GraphQL\Error\Error` for this.

Comment (calvera) [reply src/Error/Error.php]: good point, i did not notice error formatter, thanks!

**Pull request closed without merging**